### PR TITLE
Fix JmxConnectionFactory.connectAny(..) to loop a second time trying all hosts regardless of past successes/failures

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java [hunk 1]
```diff
@@ -86,17 +86,14 @@ public final JmxProxy connectAny(
...
-      throw new ReaperException("no hosts given for connectAny");
-    }
+    Preconditions.checkArgument(null != hosts && !hosts.isEmpty(), "no hosts provided to connectAny");
+
     List<String> hostList = new ArrayList<>(hosts);
     Collections.shuffle(hostList);
-    Iterator<String> hostIterator = hostList.iterator();
 
     for (int i = 0; i < 2; i++) {
-      while (hostIterator.hasNext()) {
```
Comment (michaelsembwever) [inline src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java:97]: this iterator was not being "reset" for the second loop, so the second loop never did anything…

**Review by adejanovski — APPROVED** [review]
Sorry about that bug :D please merge

File: src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java [hunk 2]
```diff
@@ -86,31 +94,24 @@ public final JmxProxy connectAny(
...
-        String host = hostIterator.next();
-
+      for (String host : hostList) {
+        assert null != host; // @todo remove the null check in the following if condition
```
Comment (tlpsonarqube) [inline src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java:104]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

**Comment by tlpsonarqube** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Review by adejanovski — APPROVED** [review]
LGTM. Let's merge

**Pull request merged**